src/Middleware/LoginSessionController.php

```diff
@@ -36,36 +33,51 @@ class LoginSessionController extends LeftAndMain
      * @param HTTPRequest $request
      * @return HTTPResponse
      */
-    public function removeLoginSession(HTTPRequest $request): HTTPResponse
+    public function remove(HTTPRequest $request): HTTPResponse
     {
-        // Ensure CSRF protection
-        if (!SecurityToken::inst()->checkRequest($request)) {
-            return $this->jsonResponse(
-                ['errors' => 'Request timed out, please try again'],
-                400
-            );
-        }
+        return $this->removeLoginSession($request);
+    }
 
-        $id = $request->param('ID');
-        $loginSession = LoginSession::get()->byID($id);
-        if (!$loginSession) {
-            return $this->jsonResponse(
-                ['errors' => 'Something went wrong.'],
-                400
-            );
-        }
+    private function removeLoginSession(HTTPRequest $request): HTTPResponse
+    {
+        $failureMessage = _t(__CLASS__ . '.REMOVE_FAILURE', 'Something went wrong.');
+        try {
+            // Ensure CSRF protection
+            if (!SecurityToken::inst()->checkRequest($request)) {
+                return $this->jsonResponse([
+                    'success' => false,
+                    'message' => $failureMessage
+                ]);
+            }
+
+            $id = $request->param('ID');
+            $loginSession = LoginSession::get()->byID($id);
+            if (!$loginSession) {
+                return $this->jsonResponse([
+                    'success' => false,
+                    'message' => $failureMessage
+                ]);
+            }
 
-        if (!$loginSession->canDelete()) {
-            return $this->jsonResponse(
-                ['errors' => 'You do not have permission to delete this record.'],
-                400
-            );
+            if (!$loginSession->canDelete()) {
+                $message = _t(__CLASS__ . '.REMOVE_PERMISSION', 'You do not have permission to delete this record.');
+                return $this->jsonResponse([
```

Comment:
This should have a 403 response code

Reply:
Status codes to be dealt with in [new issue](https://github.com/silverstripe/silverstripe-session-manager/issues/45)